event-svc/src/event/validator/event.rs

```diff
@@ -133,46 +136,43 @@ impl EventValidator {
         event_access: Arc<EventAccess>,
         ethereum_rpc_providers: Vec<ChainInclusionProvider>,
     ) -> Result<Self> {
-        let time_event_verifier = TimeEventValidator::new_with_providers(ethereum_rpc_providers);
+        let time_event_validator = TimeEventValidator::new_with_providers(ethereum_rpc_providers);
 
         Ok(Self {
             event_access,
-            time_event_verifier,
+            time_event_validator,
         })
     }
 
     /// Validates the events with the given validation requirement
-    /// If the [`ValidationRequirement`] is None, it just returns every event as valid
+    /// Regardless of the validation requirement, time events are always validated.
+    /// If the [`ValidationRequirement`] is None, it just returns every data event as valid.
     pub(crate) async fn validate_events(
         &self,
         validation_req: Option<&ValidationRequirement>,
         parsed_events: Vec<UnvalidatedEvent>,
     ) -> Result<ValidatedEvents> {
-        let validation_req = if let Some(req) = validation_req {
-            req
-        } else {
-            // we don't validate so we just return done
-            return Ok(ValidatedEvents {
-                valid: parsed_events
-                    .into_iter()
-                    .map(ValidatedEvent::from_unvalidated_unchecked)
-                    .collect(),
-                unvalidated: Vec::new(),
-                invalid: Vec::new(),
-            });
-        };
-
         let mut validated = ValidatedEvents::new_with_expected_valid(parsed_events.len());
-        // partition the events by type of validation needed and delegate to validators
+
+        // Partition the events by type of validation needed and delegate to validators
         let grouped = GroupedEvents::from(parsed_events);
 
-        let (validated_signed, validated_time) = try_join!(
```

Comment:
not worth changing now, but maybe add a TODO to make data and time event validation concurrent again.  Since time event validation is mostly network i/o bound and data event validation as mostly CPU bound, they should parallelize pretty effectively.